src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+        func_80AEB264(this, &D_060097B8, 0, 0, 0);
+        thisx->flags &= ~0x10;
+        this->action = 44;
+        this->drawConfig = 1;
+    } else {
+        Actor_Kill(thisx);
+    }
+}
+
+s32 func_80AEFDC0(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    s16 temp_ret;
+
+    if (!func_8002F194(thisx, globalCtx)) {
+        thisx->flags |= 9;
+        temp_ret = Text_GetFaceReaction(globalCtx, 0x1F);
+        thisx->textId = temp_ret;
+        if ((temp_ret & 0xFFFF) == 0) {
+            thisx->textId = 0x402C;
+        }
+        func_8002F2F4(thisx, globalCtx);
+        return 0;
+    }
+    return 1;
+}
+
+s32 func_80AEFE38(EnRu1* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+        this->actor.flags &= ~0x9;
+        return 1;
+    }
+    return 0;
+}
+
+void func_80AEFE84(EnRu1* this, GlobalContext* globalCtx, s32 cond) {
+    if (cond) {
+        this->action = 45;
+    }
+}
+
+void func_80AEFE9C(EnRu1* this, GlobalContext* globalCtx) {
+    if (func_80AEFE38(this, globalCtx)) {
+        this->action = 44;
+    }
+}
+
+void func_80AEFECC(EnRu1* this, GlobalContext* globalCtx) {
+    func_80AEEF68(this, globalCtx);
+    EnRu1_FrameUpdateMatrix(this);
+    func_80AEAE1C(this);
+    func_80AEAC10(this, globalCtx);
+    func_80AEAECC(this, globalCtx);
+    func_80AEFE84(this, globalCtx, func_80AEFDC0(this, globalCtx));
+}
+
+void func_80AEFF40(EnRu1* this, GlobalContext* globalCtx) {
+    func_80AEEFEC(this, globalCtx);
+    EnRu1_FrameUpdateMatrix(this);
+    func_80AEAE1C(this);
+    func_80AEAECC(this, globalCtx);
+    func_80AEFE9C(this, globalCtx);
+}
+
+void func_80AEFF94(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    s8 actorRoom;
+
+    if ((gSaveContext.inf_table[20] & 2) && (gSaveContext.inf_table[20] & 1) &&
+        (!(gSaveContext.inf_table[20] & 0x20)) && (!(func_80AEB020(this, globalCtx)))) {
+        func_80AEB264(this, &D_060097B8, 0, 0, 0);
+        actorRoom = thisx->room;
+        this->action = 22;
+        thisx->room = -1;
+        this->drawConfig = 0;
+        this->roomNum1 = actorRoom;
+        this->roomNum3 = actorRoom;
+        this->roomNum2 = actorRoom;
+        osSyncPrintf("スイッチルトセット!!!!!!!!!!!!!!!!!!!!!!\n");
+    } else {
+        osSyncPrintf("スイッチルトセットしない!!!!!!!!!!!!!!!!!!!!!!\n");
+        Actor_Kill(thisx);
+    }
+}
+
+void func_80AF0050(EnRu1* this, GlobalContext* globalCtx) {
+    func_80AEB264(this, &D_060097B8, 0, 0, 0);
+    this->action = 36;
+    this->roomNum1 = this->actor.room;
+    this->unk_28C = func_80AEB088(globalCtx);
+    this->actor.room = -1;
+}
+
+void EnRu1_Update(EnRu1* this, GlobalContext* globalCtx) {
+    if (this->action < 0 || this->action >= 46 || D_80AF193C[this->action] == NULL) {
+        osSyncPrintf(VT_FGCOL(RED) "メインモードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+        return;
+    }
+    D_80AF193C[this->action](this, globalCtx);
```

Comment:
else maybe, though early return makes sense in this context since its an error check kinda deal

Reply:
yeah I'd prefer to keep error returns as early returns without the else..of course only if we do it consistently.